### ISSUE TITLE
Operator `gt` working as `ge`

### ISSUE THREAD

**Issue by aryzing** [dgraph-io/dgraph#2085]
If you suspect this could be a bug, follow the template.

- What version of Dgraph are you using?
```
Dgraph version   : v1.0.2
Commit SHA-1     : fb2d3d3
Commit timestamp : 2018-01-17 15:40:00 +1100
Branch           : release/v1.0.2
```

- Have you tried reproducing the issue with latest release?
yes

- What is the hardware spec (RAM, OS)?
16GB, Ubuntu 16.04 LTS

- Steps to reproduce the issue (command/config used to run Dgraph).
Issue the following query using data from the second section of the tour, "Basic GraphQL+- Features"

```
{
  lots_of_friends(func: gt(age, 19), orderasc: age) {
    name@.
    age
  }
}
```

- Expected behaviour and actual result.
**Expected**: Catalina, aged 19, should not be in the results array
**Actual**: Catalina is included in the results array,
 
![gt-bug](https://user-images.githubusercontent.com/4924405/35768557-daf3236e-08fd-11e8-8a06-646752341ae0.png)

**Comment by demamoh** [comment]
I have also confirmed the same is happening with the lt and le operators. If you try to do 
```
{
  friends(func: allofterms(name, "Michael")) {
    name
    age
    friend @filter(ge(age, 27) AND lt(age, 35)) {
      name@.
      age
    }
  }
}
```
It returns results with age equal to 35.

**Comment by sboorlagadda** [comment]
Fixed in v1.0.3
